dnfdaemon-server/main.cpp

```diff
@@ -22,17 +22,40 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 #include "session_manager.hpp"
 
 #include <sdbus-c++/sdbus-c++.h>
+#include <signal.h>
 
 #include <iostream>
 
-static std::unique_ptr<SessionManager> session_manager = nullptr;
+static std::mutex session_manager_ptr_mutex;
+static std::atomic<SessionManager *> session_manager_ptr{nullptr};
```

Comment:
Since you ended up using a mutex to synchronize (which is a good idea), the `std::atomic` is redundant, I'd remove it.